File: app/src/main/java/cn/surine/schedulex/ui/schedule_init/ScheduleInitFragment.java
package cn.surine.schedulex.ui.schedule_init;


import android.annotation.SuppressLint;
import android.os.Bundle;
import android.text.TextUtils;

import androidx.lifecycle.ViewModelProviders;

import cn.surine.schedulex.R;
import cn.surine.schedulex.base.controller.BaseBindingFragment;
import cn.surine.schedulex.base.utils.Navigations;
import cn.surine.schedulex.base.utils.Toasts;
import cn.surine.schedulex.databinding.FragmentScheduleInitBinding;

public class ScheduleInitFragment extends BaseBindingFragment<FragmentScheduleInitBinding> {

    private ScheduleInitViewModel scheduleInitViewModel;

    public static final String SCHEDULE_NAME = "SCHEDULE_NAME";

    public static ScheduleInitFragment newInstance() {
        return new ScheduleInitFragment();
    }

    @Override
    public int layoutId() {
        return R.layout.fragment_schedule_init;
    }

    @Override
    protected void onInit(FragmentScheduleInitBinding t) {
        scheduleInitViewModel = ViewModelProviders.of(this).get(ScheduleInitViewModel.class);
        t.setData(scheduleInitViewModel);
        t.welcome.setOnClickListener(v -> {
            if (TextUtils.isEmpty(scheduleInitViewModel.scheduleName.getValue())) {
                Toasts.toast(getString(R.string.param_empty));
                return;
            }
            //传递数据给DataFetch
            Bundle bundle = new Bundle();
            bundle.putString(SCHEDULE_NAME, scheduleInitViewModel.scheduleName.getValue());
            Navigations.open(ScheduleInitFragment.this, R.id.action_scheduleInitFragment_to_dataFetchFragment, bundle);
        });
    }

    @SuppressLint("MissingSuperCall")
    @Override
    public void onBackPressed() {
        activity().finish();
    }
}

File: app/src/main/java/cn/surine/schedulex/ui/schedule_init/ScheduleInitFragment.kt
package cn.surine.schedulex.ui.schedule_init

import android.annotation.SuppressLint
import android.os.Bundle
import androidx.lifecycle.ViewModelProviders
import cn.surine.schedulex.R
import cn.surine.schedulex.base.controller.BaseBindingFragment
import cn.surine.schedulex.base.utils.Navigations
import cn.surine.schedulex.base.utils.Navigations.open
import cn.surine.schedulex.base.utils.Toasts.toast
import cn.surine.schedulex.databinding.FragmentScheduleInitBinding
import cn.surine.schedulex.third_parse.Shell
import kotlinx.android.synthetic.main.fragment_schedule_init.*

/**
 * Intro：
 *
 * @author sunliwei
 * @date 2020/6/24 16:06
 */
class ScheduleInitFragment : BaseBindingFragment<FragmentScheduleInitBinding>() {
    private lateinit var scheduleInitViewModel: ScheduleInitViewModel

    companion object {
        const val SCHEDULE_NAME = "SCHEDULE_NAME"
        const val PAGE_FLAG = "PAGE_FLAG"
    }

    override fun layoutId(): Int = R.layout.fragment_schedule_init

    override fun onInit(t: FragmentScheduleInitBinding) {
        scheduleInitViewModel = ViewModelProviders.of(this)[ScheduleInitViewModel::class.java]
        t.data = scheduleInitViewModel
        welcome.setOnClickListener {
            if (scheduleInitViewModel.scheduleName.value != null) {
                open(this, R.id.action_scheduleInitFragment_to_dataFetchFragment, Bundle().apply {
                    putString(SCHEDULE_NAME, scheduleInitViewModel.scheduleName.value)
                })
            } else {
                toast(getString(R.string.param_empty))
            }
        }
    }

    @SuppressLint("MissingSuperCall")
    override fun onBackPressed() {
        if (arguments?.getBoolean(PAGE_FLAG, false) == true) {
            Navigations.close(this)
        } else {
            activity().finish()
        }
    }

}